modules/discoveryBidAdapter.js

```diff
@@ -0,0 +1,525 @@
+        '|oran|owg1|p800|pan(a|d|t)|pdxg|pg(13|-([1-8]|c))|phil|pire|pl(ay|uc)|pn-2|po(ck|rt|se)|prox|psio',
+        '|pt-g|qa-a|qc(07|12|21|32|60|-[2-7]|i-)|qtek|r380|r600|raks|rim9|ro(ve|zo)|s55/|sa(ge|ma|mm|ms',
+        '|ny|va)|sc(01|h-|oo|p-)|sdk/|se(c(-|0|1)|47|mc|nd|ri)|sgh-|shar|sie(-|m)|sk-0|sl(45|id)|sm(al',
+        '|ar|b3|it|t5)|so(ft|ny)|sp(01|h-|v-|v )|sy(01|mb)|t2(18|50)|t6(00|10|18)|ta(gt|lk)|tcl-|tdg-|tel(i|m)',
+        '|tim-|t-mo|to(pl|sh)|ts(70|m-|m3|m5)|tx-9|up(.b|g1|si)|utst|',
+        'v400|v750|veri|vi(rg|te)|vk(40|5[0-3]|-v)',
+        '|vm40|voda|vulc|vx(52|53|60|61|70|80|81|83|85|98)|w3c(-| )|webc|whit|wi(g |nc|nw)|wmlb|wonu|x700|yas-',
+        '|your|zeto|zte-',
+      ].join(''),
+      'i'
+    );
+    if (reg1.test(a) || reg2.test(a.substr(0, 4))) {
+      check = true;
+    }
+  })(navigator.userAgent || navigator.vendor || window.opera);
+  return check;
+}
+
+/**
+ * get BidFloor
+ * @param {*} bid
+ * @param {*} mediaType
+ * @param {*} sizes
+ * @returns
+ */
+function getBidFloor(bid) {
+  if (!utils.isFn(bid.getFloor)) {
+    return utils.deepAccess(bid, 'params.bidfloor', 0);
+  }
+
+  try {
+    const bidFloor = bid.getFloor({
+      currency: 'USD',
+      mediaType: '*',
+      size: '*',
+    });
+    return bidFloor.floor;
+  } catch (_) {
+    return 0;
+  }
+}
+
+/**
+ * get sizes for rtb
+ * @param  {Array|Object} requestSizes
+ * @return {Object}
+ */
+function transformSizes(requestSizes) {
+  let sizes = [];
+  let sizeObj = {};
+
+  if (
+    utils.isArray(requestSizes) &&
+    requestSizes.length === 2 &&
+    !utils.isArray(requestSizes[0])
+  ) {
+    sizeObj.width = parseInt(requestSizes[0], 10);
+    sizeObj.height = parseInt(requestSizes[1], 10);
+    sizes.push(sizeObj);
+  } else if (typeof requestSizes === 'object') {
+    for (let i = 0; i < requestSizes.length; i++) {
+      let size = requestSizes[i];
+      sizeObj = {};
+      sizeObj.width = parseInt(size[0], 10);
+      sizeObj.height = parseInt(size[1], 10);
+      sizes.push(sizeObj);
+    }
+  }
+
+  return sizes;
+}
+
+// Support sizes
+const popInAdSize = [
+  { w: 300, h: 250 },
+  { w: 300, h: 600 },
+  { w: 728, h: 90 },
+  { w: 970, h: 250 },
+  { w: 320, h: 50 },
+  { w: 160, h: 600 },
+  { w: 320, h: 180 },
+  { w: 320, h: 100 },
+  { w: 336, h: 280 },
+];
+
+/**
+ * get aditem setting
+ * @param {Array}  validBidRequests an an array of bids
+ * @param {Object} bidderRequest  The master bidRequest object
+ * @return {Object}
+ */
+function getItems(validBidRequests, bidderRequest) {
+  let items = [];
+  items = validBidRequests.map((req, i) => {
+    let ret = {};
+    // eslint-disable-next-line no-debugger
+    let mediaTypes = getKv(req, 'mediaTypes');
+
+    const bidFloor = getBidFloor(req);
+    let id = '' + (i + 1);
+
+    if (mediaTypes.native) {
+      ret = {
+        id: id,
+        bidfloor: bidFloor,
+        // TODO Dynamic parameters
+        native: {
```

Comment:
Lets set this native object in a const and reference that here to shorten the function length